internal/config/config.go

```diff
@@ -368,6 +386,19 @@ func New() *Config {
 	}
 }
 
+// SetupUser performs late initialization of user-specific configuration,
+// determining the current user. This is in a separate method as it
+// can fail, and is called after flag parsing.
+func (c *Config) SetupUser() error {
```

Comment:
since this gets called inside librarian. Run, I would remove this and move it to `New`:

```
func New() (*Config, error) {
	currentUser, err := user.Current()
	if err != nil {
		return fmt.Errorf("failed to get current user: %w", err)
	}
        ...
}
```